emukit/bayesian_optimization/acquisitions/expected_improvement.py

```diff
@@ -75,6 +75,43 @@ def has_gradients(self) -> bool:
         """Returns that this acquisition has gradients"""
         return isinstance(self.model, IDifferentiable)
 
+    def _get_model_predictions(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """Get model predictions for the function values at given input locations."""
+        return self.model.predict(x)
+
+    def _get_y_minimum(self) -> np.ndarray:
+        """Return the minimum value in the samples observed so far."""
+        return np.min(self.model.Y, axis=0)
+
+
+class MeanPluginExpectedImprovement(ExpectedImprovement):
+    def __init__(self, model: IModelWithNoise, jitter: float=0.0)-> None:
+        """
+        This acquisition computes for a given input the expected improvement over the current best mean at one of the
+        observed in inputs.
+
+        This is a heuristic that allows Expected Improvement to deal with problems with noisy observations, where
+        the standard Expected Improvement might fail if the noise is too large.
+
+        For more information see:
+            "A benchmark of kriging-based infill criteria for noisy optimization" by Picheny et al. 
+        Note: the model type should be Union[IPredictsWithNoise, Intersection[IpredictsWithNoise, IDifferentiable]].
+            Support for Intersection types might be added to Python in the future (see PEP 483)
+
+        :param model: model that is used to compute the improvement.
+        :param jitter: parameter to encourage extra exploration.
+        """
+        super().__init__(model=model, jitter=jitter)
+
+    def _get_y_minimum(self):
+        """Return the smallest model mean prediction at the previously observed points."""
+        means_at_prev, _ = self.model.predict_noiseless(self.model.X)
+        return np.min(means_at_prev, axis=0)
```

Comment:
Just curious, are these two lines all there is to the heuristic?

Reply:
Yes, I think so

Reply:
It's a very heuristical heuristic